app/questionnaire/questionnaire_schema.py

```diff
@@ -293,6 +304,23 @@ def _update_answer_dependencies_for_calculated_summary(
                 self._get_answer_dependent_for_block_id(block_id=block_id)
             }
 
+    def _update_answer_dependencies_for_grand_calculated_summary(
+        self,
+        grand_calculated_summary_calculated_summary_ids: Iterable[str],
+        block_id: str,
+    ) -> None:
+        # validator ensures that any grand calculated summary will come after the calculated summaries it depends on
+        # so the calculated summary dependencies will already be done, and this can just take those
+        for calculated_summary_id in grand_calculated_summary_calculated_summary_ids:
+            # Type ignore: safe to assume block exists
+            calculated_summary_block: ImmutableDict = self.get_block(calculated_summary_id)  # type: ignore
+            for answer_id in get_calculated_summary_answer_ids(
+                calculated_summary_block
+            ):
+                self._answer_dependencies_map[answer_id] |= {
+                    self._get_answer_dependent_for_block_id(block_id=block_id)
+                }
```

Comment:
Instead of duplicating this, I suggest reusing `_update_answer_dependencies_for_calculated_summary` here.

Could even move getting the answer ids for the block into that method.

Reply:
added the dependent block to that function so it can be reused 👍